float32 width
bool moving
bool grippedDetected
